### DISCUSSION TITLE
Inject tailwind props into jsx elements

### DISCUSSION THREAD

**Discussion by mtbaqer** [ben-rogerson/twin.macro#501] [Ideas]
I was thinking just like how we are injecting `tw` prop into jsx elements, would it be possible for us to directly inject the tailwind props into the elements? 

Turning elements like this:
```
<input tw="border hover:border-black" />
```
to something like this:
```
<input border hover:border-black />
```

Specially given the new Typescript 4.4 [template literal string index signatures](https://github.com/microsoft/TypeScript/pull/44512) to resolve all the types problems. 

It would result, in my opinion, to a cleaner, more readable alternative than using strings or a combination of template literals and editor plugins.

It would also leverage Typescript's type errors detections instead of relying on something else.

Thoughts?

**Comment by ben-rogerson** [comment]
I think it's probably possible but I'm not sure it could be considered a "cleaner" alternative as your styles would be dispersed among the other element props as there's no "container" for them.
Also, spaces wouldn't be allowed so features like groups wouldn't be supported in their current form.
